tests/Doctrine/ODM/MongoDB/Tests/Functional/PropertyHooksTest.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ODM\MongoDB\Tests\Functional;
+
+use Doctrine\ODM\MongoDB\Mapping\MappingException;
+use Doctrine\ODM\MongoDB\Tests\BaseTestCase;
+use Documents84\PropertyHooks\MappingVirtualProperty;
+use Documents84\PropertyHooks\User;
+use PHPUnit\Framework\Attributes\RequiresPhp;
+
+#[RequiresPhp('>= 8.4.0')]
+class PropertyHooksTest extends BaseTestCase
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        if ($this->dm->getConfiguration()->isNativeLazyObjectsEnabled()) {
+            return;
+        }
+
+        $this->markTestSkipped('Property hooks require native lazy objects to be enabled.');
```

Comment:
Did phpcs ask for the early return? I find the following easier to read...

```suggestion
        if ($this->dm->getConfiguration()->isNativeLazyObjectsEnabled()) {
            $this->markTestSkipped('Property hooks require native lazy objects to be enabled.');
        }
```

Reply:
Yes, PHPCS. I hate this rule. We need a way to disable it when the condition only contains 1 statement.

Reply:
The [EarlyExit sniff](https://github.com/slevomat/coding-standard/blob/master/doc/control-structures.md#slevomatcodingstandardcontrolstructuresearlyexit-) supports the `ignoreStandaloneIfInScope`, `ignoreOneLineTrailingIf`, and `ignoreTrailingIfWithOneInstruction` options that allow us to customise this. I believe the `ignoreTrailingIfWithOneInstruction` option allows this kind of usage, but I also don't mind enabling all of them.